File: androidmvp/src/main/java/com/hsc/vince/androidmvp/base/MvpFragment.java
package com.hsc.vince.androidmvp.base;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;

/**
 * <p>作者：黄思程  2018/4/12 10:15
 * <p>邮箱：codinghuang@163.com
 * <p>作用：
 * <p>描述：带网络请求的Fragment
 */
public abstract class MvpFragment<P extends BasePresenter> extends BaseFragment {
    protected P mPresenter;

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        mPresenter = createPresenter();
    }

    protected abstract P createPresenter();

    @Override
    public void onDestroy() {
        super.onDestroy();

        if (mPresenter != null) {
            mPresenter.detachView();
        }
    }
}

File: news/src/main/java/com/vince/news/base/MvpFragment.kt
package com.vince.news.base

import android.os.Bundle
import android.view.View

/**
 * <p>作者：黄思程  2017/11/16 17:00
 * <p>邮箱：huangsicheng@gouuse.cn
 * <p>作用：
 * <p>描述：带网络请求的Fragment
 */
abstract class MvpFragment<P : BasePresenter<*>> : BaseFragment() {
    protected lateinit var mvpPresenter: P

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        mvpPresenter = createPresenter()

        super.onViewCreated(view, savedInstanceState)
    }

    override fun onDestroyView() {
        super.onDestroyView()

        mvpPresenter.detachView()
    }

    /**
     * @return 创建Presenter
     * */
    protected abstract fun createPresenter(): P
}